GUIDELINES.md

```diff
@@ -0,0 +1,135 @@
+# sweetest test development guidelines
+
+After some time working with sweetest we came up to the conclusion that it leaves a lot of freedom to design tests as we want. But on the other side there is an increasing need for alignment. These guidelines are here for reaching an appropriate level of alignment. Feel free to challenge the current state and to contribute!
+
+## Content
+
+* [Goals](#goals)
+* [One-by-one steps](#one-by-one-steps)
+* [Principles](#principles)
+* [Links](#links)
+
+## Goals
+
+* Put a **layer of abstraction** on the system under test: we call these abstractions _steps_ (as derived from Cucumber)
+  * so if the system under test changes, the test system doesn't need to change
+  * so the test just tells _what_ is tested, not _how_ (tests become more business-centric, whenever possible, all the details are in the steps)
+  * so you can reuse test code
+  * to sum up, there is a proper separation of concerns (separation of business rules, workflow and technical implementation) and the ability to reuse test code
+* **Simplify dependency tree creation** by using configuration and automatic dependency resolution
+* That **simplifies test setup** which makes it **cheaper to have bigger integration tests**
+* With that in mind it becomes worth to **invest more in integration tests** instead of unit tests
+  * which **reduces the use of mocks** and
+  * leads to **more realistic tests**
```

Comment:
But it also increases coupling between tests, right?
e.g. if I change one steps class, it could influence a lot of other step classes.

Reply:
that's right, so there's pros and cons. it becomes just more important to do abstraction right. do you see a big impendiment?